app/bootloader/main.cpp

```diff
@@ -1,55 +1,56 @@
-/*
- * Copyright (c) 2020 Embedded Planet
- * Copyright (c) 2020 ARM Limited
- * SPDX-License-Identifier: Apache-2.0
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License
- */
+// Leka - LekaOS
+// Copyright 2020 Embedded Planet
+// Copyright 2020 ARM Limited
+// SPDX-License-Identifier: Apache-2.0
+// Original file: https://github.com/mcu-tools/mcuboot/blob/main/boot/mbed/mcuboot_main.cpp
 
 #include "mbed_application.h"
 #include "mbedtls/platform.h"
 
 #include "drivers/BufferedSerial.h"
+#include "drivers/DigitalIn.h"
+#include "rtos/ThisThread.h"
 
+#include "CoreBattery.h"
 #include "LogKit.h"
 #include "bootutil/bootutil.h"
 #include "bootutil/image.h"
-#include "hal/serial_api.h"
 
-static auto serial = mbed::BufferedSerial(USBTX, USBRX, 115200);
+using namespace std::chrono;
+using namespace leka;
+
+static auto serial		 = mbed::BufferedSerial {CONSOLE_TX, CONSOLE_RX, 115200};
+auto charge_status_input = mbed::DigitalIn {PinName::BATTERY_CHARGE_STATUS};
+auto battery			 = leka::CoreBattery {PinName::BATTERY_VOLTAGE, charge_status_input};
```

Comment:
ça change quoi si on met tout en static?

```suggestion
static auto charge_status_input = mbed::DigitalIn {PinName::BATTERY_CHARGE_STATUS};
static auto battery			 = leka::CoreBattery {PinName::BATTERY_VOLTAGE, charge_status_input};
```

Reply:
C'est l'objet de cette PR, résultat ici: https://github.com/leka/LekaOS/pull/389#issuecomment-988985860